src/components/NewDiskDialog/index.js

```diff
@@ -0,0 +1,315 @@
+  }
+
+  /**
+   * active in VM's datacenter, data type
+   */
+  getSuitableStorageDomains () {
+    const dataCenterId = this.getDataCenterId()
+    const storageDomains = Object.values(this.props.storageDomains)
+      .filter(storageDomain =>
+        storageDomain.statusPerDataCenter[dataCenterId] === 'active' &&
+        storageDomain.type === 'data')
+    return storageDomains
+  }
+
+  getAlreadyUsedSuitableStorageDomainId () {
+    if (!this.props.vm.disks) {
+      return undefined
+    }
+    const usedStorageDomainIds = flatMap(Object.values(this.props.vm.disks),
+      disk => disk.storageDomainId ? [ disk.storageDomainId ] : [])
+    for (let suitableStorageDomain of this.getSuitableStorageDomains()) {
+      if (usedStorageDomainIds.includes(suitableStorageDomain.id)) {
+        return suitableStorageDomain.id
+      }
+    }
+    return undefined
+  }
+
+  getPreselectedStorageDomainId () {
+    const alreadyUsed = this.getAlreadyUsedSuitableStorageDomainId()
+    if (alreadyUsed) {
+      return alreadyUsed
+    }
+    const suitableStorageDomains = this.getSuitableStorageDomains()
+    return suitableStorageDomains[0] && suitableStorageDomains[0].id
+  }
+
+  getNextAliasSuggestion () {
+    const diskNamePrefix = this.props.vm.name + '_Disk'
+    const aliasNumbers = this.props.vm.disks
+      ? this.props.vm.disks
+        .map(disk => disk.name)
+        .filter(alias => alias.startsWith(diskNamePrefix))
+        .map(alias => Number.parseInt(alias.substr(diskNamePrefix.length)))
+        .filter(numericalSuffix => !Number.isNaN(numericalSuffix))
+      : []
+    if (aliasNumbers.length === 0) {
+      return diskNamePrefix + '1'
+    }
+    const maximalIndex = Math.max(...aliasNumbers)
+    return diskNamePrefix + (maximalIndex + 1)
+  }
+
+  getDataCenterName () {
+    const dataCenterId = this.getDataCenterId()
+    const dataCenter = this.props.dataCenters
+      .filter(dataCenter => dataCenter.id === dataCenterId)[0]
+    if (!dataCenter) {
+      return '[unknown]'
+    }
+    return dataCenter.name
+  }
+
+  /**
+   * http://ovirt.github.io/ovirt-engine-api-model/master/#types/disk_interface
+   */
+  getInterface () {
+    const disksOnSelectedStorageDomain = this.state.storageDomainId &&
+      this.props.vm.disks &&
+      this.props.vm.disks.filter(disk => disk.storageDomainId === this.state.storageDomainId)
+    if (!disksOnSelectedStorageDomain || disksOnSelectedStorageDomain.length === 0) {
+      return 'virtio_scsi'
+    }
+    return disksOnSelectedStorageDomain[0].iface
+  }
+
+  onAliasChange (event) {
+    this.setState({ alias: event.target.value })
+  }
+
+  onSizeChange (event) {
+    this.setState({ sizeGb: event.target.value })
+  }
+
+  onStorageDomainChange (selectedStorageDomainId) {
+    this.setState({ storageDomainId: selectedStorageDomainId })
+  }
+
+  onCancel () {
+    this.props.onClose()
+  }
+
+  getSizeBytes () {
+    const parsedGb = Number.parseInt(this.state.sizeGb)
```

Comment:
optional: can we introduce a convertor function? In helpers or so.

Reply:
done